api/src/main/java/org/openmrs/messagesource/impl/MessageSourceServiceImpl.java

```diff
@@ -96,7 +96,12 @@ public void setActiveMessageSource(MutableMessageSource activeMessageSource) {
 	 *      java.util.Locale)
 	 */
 	public String getMessage(MessageSourceResolvable resolvable, Locale locale) {
-		return activeMessageSource.getMessage(resolvable, locale);
+		if((resolvable.getCodes()[0]).equals((activeMessageSource.getMessage(resolvable, locale)))){
```

Comment:
Apparently it is, I've always guessed that this is a spring bug because we never do anything with resolving the code in our message source implementation, you can clearly see [here](http://grepcode.com/file/repository.springsource.com/org.springframework/org.springframework.context/3.0.3/org/springframework/context/support/AbstractMessageSource.java#155) that spring is returning the first message code which is contrary to what they preach in their javadocs